### PR TITLE
Implement cubic spline class

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This PR introduces the `Spline` class that is an abstract class. `QuinticSpline` and `CubicSpline` inherit from `Spline`. 

I finally modify the SwingFoot planner, so the user can choose between the two spline implementations.

### CODE REVIEW

File: src/Planners/src/CubicSpline.cpp [hunk 1]
```diff
@@ -0,0 +1,768 @@
...
+    {
+        Eigen::VectorXd position; /**< Knot position */
+        Eigen::VectorXd velocity; /**< first derivative of spline computed at t@knot */
+        Eigen::VectorXd acceleration; /**< second derivative of spline computed at t@knot */
+
+        double timeInstant; /**< Knot time (it is an absolute time ) */
+    };
+
+    /**
+     * Description of a 5-th polynomial. It contains the coefficients, the initial/final knots and
```
Comment (paolo-viceconte) [inline src/Planners/src/CubicSpline.cpp:42]: Another 5-th order polynomial to be replaced with 3-rd

File: src/Planners/src/CubicSpline.cpp [hunk 2]
```diff
@@ -0,0 +1,768 @@
+/**
+ * @file CubicSpline.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
```
Comment (paolo-viceconte) [inline src/Planners/src/CubicSpline.cpp:4]: Another 2020 to be replaced here

File: src/Planners/src/CubicSpline.cpp [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+
+CubicSpline::CubicSpline()
+{
+    m_pimpl = std::make_unique<Impl>();
+}
+
+CubicSpline::~CubicSpline() = default;
+
+bool CubicSpline::setInitialConditions(Eigen::Ref<const Eigen::VectorXd> velocity,
+                                       Eigen::Ref<const Eigen::VectorXd> /**acceleration*/)
```
Comment (paolo-viceconte) [inline src/Planners/src/CubicSpline.cpp:183]: Not sure whether the second parameter is left on purpose here. If so, how is it exploited?

File: src/Planners/src/CubicSpline.cpp [hunk 4]
```diff
@@ -0,0 +1,768 @@
...
+    m_pimpl->currentTrajectory.acceleration = Eigen::VectorXd::Zero(velocity.size());
+
+    // The initial conditions changed. The coefficients are outdated.
+    m_pimpl->areCoefficientsComputed = false;
+
+    return true;
+}
+
+bool CubicSpline::setFinalConditions(Eigen::Ref<const Eigen::VectorXd> velocity,
+                                     Eigen::Ref<const Eigen::VectorXd> /**acceleration*/)
```
Comment (paolo-viceconte) [inline src/Planners/src/CubicSpline.cpp:204]: Same as above

**Review by paolo-viceconte — APPROVED** [review]
It looks good to me, I just spotted a few potential fixes in the documentation. I will try to exploit this class soon, thanks @GiulioRomualdi!

File: src/Planners/src/CubicSpline.cpp (hunk 3 above)
Comment (GiulioRomualdi) [reply src/Planners/src/CubicSpline.cpp:183]: `acceleration` is not used however is required to keep the same interface between the two splines

**Pull request merged**